versioncontrol/peer.go

```diff
@@ -178,8 +218,64 @@ func (peer *Peer) Close() (err error) {
 // Addr returns the public address.
 func (peer *Peer) Addr() string { return peer.Server.Listener.Addr().String() }
 
-func configToProcess(binary Binary) version.Process {
-	return version.Process{
+// ValidateRollouts validates the rollout field of each field in the Versions struct.
+func (versions Versions) ValidateRollouts(log *zap.Logger) error {
+	value := reflect.ValueOf(versions)
+	fieldCount := value.NumField()
+	validationErrs := errs.Group{}
+	for i := 1; i < fieldCount; i++ {
+		binary, ok := value.Field(i).Interface().(Binary)
+		if !ok {
+			log.Warn("non-binary field in versions config struct", zap.String("field name", value.Type().Field(i).Name))
+			continue
+		}
+		if err := binary.Rollout.Validate(value.Type().Field(i).Name, log); err != nil {
+			validationErrs.Add(err)
+		}
+	}
+	return validationErrs.Err()
+}
+
+// Validate validates the rollout seed and cursor config values.
+func (rollout Rollout) Validate(binary string, log *zap.Logger) error {
+	seedLen := len(rollout.Seed)
+	if seedLen == 0 {
+		log.Warn("empty seed", zap.String("binary", binary))
```

Comment:
I think that's more idiomatic to return a different error to indicate that's `seedLen` is 0 rather than passing the logger and log a warning.